NSS-Day-Cohort-42/rare-API-media-moguls
# POST creates a subscription when given an author ID, actions get active subscriptions and add end dates

#### Changes Made
1. Modified file `subscription.py` in `views` directory to include POST functionality, plus actions to PATCH (end subscription) and return only active subscriptions.
​
#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout ap-subscriptions
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
```
..
server
```
4. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py runserver
```
5. Test app functionality.
> Pop into postman and give `    {
        "author": 2
    }` to `http://localhost:8000/subscriptions`
> Should return an object with the authed user as the follower, created_on as the current time, and a null value for ended_on
> Bonus: send it `    {
        "author_id": 1
    }` instead!
> Send a GET request to `http://localhost:8000/subscriptions/get_current_subscriptions`
>Send a PATCH to `http://localhost:8000/subscriptions/4/unsubscribe`
> Is there a more reasonable way to fetch that author ID from the author profile? Doubling up on that RareUser was giving me the runaround--this would WORK if the front-end just passes that author ID, but I feel like there's a more direct back-end route?

6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
